vmcnet/train/runners.py

```diff
@@ -615,27 +627,28 @@ def run_molecule() -> None:
         apply_pmap=config.distribute,
     )
 
-    reload_from_checkpoint = (
-        reload_config.logdir != train.default_config.NO_RELOAD_LOG_DIR
-        and reload_config.use_checkpoint_file
-    )
-
-    if reload_from_checkpoint:
+    if reload:
         checkpoint_file_path = os.path.join(
             reload_config.logdir, reload_config.checkpoint_relative_file_path
         )
         directory, filename = os.path.split(checkpoint_file_path)
-        _, data, params, optimizer_state, key = utils.io.reload_vmc_state(
-            directory, filename
-        )
-        (
-            data,
-            params,
-            optimizer_state,
-            key,
-        ) = utils.distribute.distribute_vmc_state_from_checkpoint(
-            data, params, optimizer_state, key
-        )
+
+        if reload_run_from_checkpoint:
+            _, data, params, optimizer_state, key = utils.io.reload_vmc_state(
+                directory, filename
+            )
+            (
+                data,
+                params,
+                optimizer_state,
+                key,
+            ) = utils.distribute.distribute_vmc_state_from_checkpoint(
+                data, params, optimizer_state, key
+            )
+
+        if reload_model_from_checkpoint:
+            params = utils.io.reload_vmc_state(directory, filename)[2]
+            params = jax.tree_map(lambda x: x[None], params)
```

Comment:
Does this work properly on a multi GPU setting? 

In any case can you update to use distribute.replicate_all_local_devices for consistency?